### PR TITLE
Add `catalogApiMock` to `catalog-react/testUtils`

### PR DESCRIPTION
**Pull request by freben**
Note that this also adds an `ApiMock` type to `frontend-test-utils` analogously to what the `ServiceMock` type in `backend-test-utils` type does.

The old `MockEntityListContextProvider` test utility was deprecated in the index export and moved to `/testUtils` as well.

### CODE REVIEW

File: plugins/org/src/components/Cards/OwnershipCard/useGetEntities.test.ts
Comment (freben) [inline plugins/org/src/components/Cards/OwnershipCard/useGetEntities.test.ts:1]: yeah this test isn't pretty, but it was worse before and didn't want to spend time rewriting it with nicer patterns

**Comment by freben** [comment]
There's a lot more replacing to be done in tests, but to keep the PR somewhat easy to overview, I opted to end here and do the rest in a separate PR

File: plugins/catalog-react/src/testUtils/catalogApiMock.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  createApiFactory,
+} from '@backstage/frontend-plugin-api';
+import { InMemoryCatalogClient } from '@backstage/catalog-client/testUtils';
+import { Entity } from '@backstage/catalog-model';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import { CatalogApi } from '@backstage/catalog-client';
+import { ApiMock } from '@backstage/frontend-test-utils';
+
+/** @internal */
+function simpleMock<TApi>(
```
Comment (benjdlambert) [inline plugins/catalog-react/src/testUtils/catalogApiMock.ts:29]: should we move this somewhere? feeling it's gonna be c+p'd a alot.

Comment (freben) [reply plugins/catalog-react/src/testUtils/catalogApiMock.ts:29]: Maybe next step yeah. I'll probably start moving toward `mockApis.foo` and at that point ...

Comment (Rugvip) [reply plugins/catalog-react/src/testUtils/catalogApiMock.ts:29]: yep might be that these make sense as a `/testUtils` export on the `*-plugin-api` packages themselves

**Review by Rugvip — APPROVED** [review]
Nice! 😁 🎉

**Comment by freben** [comment]
Oh come to think of it. `simpleMock` is a good candidate for an `@internal` package. We aren't really looking to create a facility for others to make mocks, to if we are afraid of too much copy-pasta, it should probably be just put in the internals.

**Pull request merged**